### ISSUE TITLE
annotation.on.supertype does not resolve annotation aliases

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#2051]
`BaseTypeVisitor`'s `annotation.on.supertype` check (`checkExtendsImplements`, gated on
`atypeFactory.isSupportedQualifier(am)`) tests an annotation by exact identity, not by resolving
aliases. So a supertype-clause annotation written with an *aliased* form of a qualifier is
silently accepted, while the canonical form is correctly rejected.

## Reproducer (Nullness Checker)

```java
import org.checkerframework.checker.nullness.qual.Nullable;

class UsesCanonical {
    // error: (annotation.on.supertype) -- correct
    static class T extends @Nullable Object {}
}
```

```java
import org.jspecify.annotations.Nullable;

class UsesJSpecifyAlias {
    // no error -- wrong; org.jspecify.annotations.Nullable is an alias for the canonical
    // org.checkerframework.checker.nullness.qual.Nullable, and this is the exact same code
    static class T extends @Nullable Object {}
}
```

The same gap applies to `implements`, an interface's own `extends`, and `implements` on an enum
or record.

## Why this matters for JSpecify

JSpecify's spec explicitly lists a class declaration's supertype as a location it gives no
meaning to (`docs/docs/spec.md`, "Other notable unrecognized annotations include: ... A type
used as a superclass or superinterface"). A user relying on `-Amode=jspecify` to reject exactly
what JSpecify rejects gets silent, incorrect acceptance whenever they write the annotation using
`org.jspecify.annotations.Nullable` -- which is the form JSpecify itself defines and the one a
JSpecify-oriented user would actually write.

## Scope

`checkExtendsImplements` is framework-level (`framework/.../common/basetype/BaseTypeVisitor.java`),
so this affects every checker with a "root type on a supertype clause is invalid" rule, not just
the Nullness Checker.

Found during a completeness audit of nullness-related type-annotation locations for PR #2024.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_019jLjeEW1F1aE2E3ax7EWqV

**Comment by aosen-xiong** [comment]
@wmdietl is this the same as #2021?

**Comment by wmdietl** [comment]
It came out of a review of #2024, so #2022 wasn't merged yet. I'll verify once 2022 is merged.
